webapp/components/announcement_bar/announcement_bar.jsx

```diff
@@ -115,28 +143,57 @@ export default class ErrorBar extends React.Component {
             e.preventDefault();
         }
 
+        if (this.state.type === BAR_ANNOUNCEMENT_TYPE) {
+            localStorage.setItem(this.state.message, true);
+        }
+
         if (ErrorStore.getLastError() && ErrorStore.getLastError().notification) {
             ErrorStore.clearNotificationError();
         } else {
             ErrorStore.clearLastError();
         }
 
-        this.setState({message: null});
+        this.setState(this.getState());
     }
 
     render() {
-        if (!this.isValidError(this.state)) {
+        if (!this.isValidState(this.state)) {
             return <div/>;
         }
 
-        var errClass = 'error-bar';
+        if (!this.props.isLoggedIn && this.state.type === BAR_ANNOUNCEMENT_TYPE) {
+            return <div/>;
+        }
 
-        if (this.state.type === BAR_DEVELOPER_TYPE) {
+        let errClass = 'error-bar';
+        const barStyle = {};
+        const linkStyle = {};
+        if (this.state.color && this.state.textColor) {
+            barStyle.backgroundColor = this.state.color;
+            barStyle.color = this.state.textColor;
+            linkStyle.color = this.state.textColor;
+        } else if (this.state.type === BAR_DEVELOPER_TYPE) {
             errClass = 'error-bar-developer';
         } else if (this.state.type === BAR_CRITICAL_TYPE) {
             errClass = 'error-bar-critical';
         }
 
+        let closeButton;
+        if (this.state.allowDismissal) {
+            closeButton = (
+                <a
+                    href='#'
+                    className='error-bar__close'
+                    style={linkStyle}
+                    onClick={this.handleClose}
+                >
+                    {'×'}
+                </a>
+            );
+        } else {
+            barStyle.position = 'static';
```

Comment:
This was temporary until @asaadmahmood made his changes. We should probably wait to dev until we're done with PM review